sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2243,23 +2243,11 @@ void SetArgBasedOnType(
         getMemAllocationFunc
             ? (sycl::detail::pi::PiMem)getMemAllocationFunc(Req)
             : nullptr;
-    if (Context.get_backend() == backend::opencl) {
-      // clSetKernelArg (corresponding to piKernelSetArg) returns an error
-      // when MemArg is null, which is the case when zero-sized buffers are
-      // handled. Below assignment provides later call to clSetKernelArg with
-      // acceptable arguments.
-      if (!MemArg)
-        MemArg = sycl::detail::pi::PiMem();
-
-      Plugin->call<PiApiKind::piKernelSetArg>(
-          Kernel, NextTrueIndex, sizeof(sycl::detail::pi::PiMem), &MemArg);
```

Comment:
This was a problem before due to direct casting, but that now should be handled in the default pass as we have a UR layer that could handle that.